app/controllers/physical_server_controller.rb

```diff
@@ -36,4 +36,68 @@ def textual_group_list
     []
   end
   helper_method :textual_group_list
+
+  # Handles buttons pressed on the toolbar
+  def button
+    # Get the list of servers to apply the action to
+    servers = retrieve_servers
+
+    # Apply the action (depending on the button pressed) to the servers
+    apply_action_to_servers(servers)
+  end
```

Comment:
The way our controllers usually implement the `button` method is, they direct to the appropriate button action based on `params[:pressed]`
For e.g.

```ruby
if params[:pressed] == 'physical_server_power_on'
  physical_server_power_on
elsif params[:pressed] == 'physical_server_power_off'
  physical_server_power_off
end
```

That's all you need to do here.